internal/show/versions.go

```diff
@@ -67,19 +94,119 @@ func printDeploymentVersions(clusterInfo *cluster.Info, printer *table.Printer,
 		}
 
 		version, repository := images.ParseOperatorImage(deployment.Spec.Template.Spec.Containers[0].Image)
-		printer.Add(component, repository, version)
+
+		runningVersion, err := getVersionForComponent(clusterInfo, component,
+			labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels).String())
+		if err != nil {
+			return err
+		}
+
+		printer.Add(component, repository, version, runningVersion)
 	}
 
 	return nil
 }
 
+func getVersionForComponent(clusterInfo *cluster.Info, component, labelSelector string) (string, error) {
+	podsClient := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Pods(constants.OperatorNamespace)
+	podList, err := podsClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+
+	if err != nil || len(podList.Items) < 1 {
+		return "", errors.Wrapf(err, "Failed to find pods for component %s", component)
+	}
+
+	// Try all pods
+	for i := range podList.Items {
+		pod := podList.Items[i]
+		podVersion, _ := getVersionFromPodBinary(&pod, clusterInfo, component)
+
+		if podVersion != "" {
+			return podVersion, nil
+		}
+
+		podVersion = getVersionFromPodLogs(&pod, podsClient, component)
+
+		if podVersion == "" {
+			continue
+		}
+
+		return podVersion, nil
+	}
+
+	return "", errors.Errorf("Failed to get version for %s", component)
+}
+
+func getVersionFromPodBinary(pod *corev1.Pod, clusterInfo *cluster.Info, component string) (string, error) {
+	cmdArg := "--version"
+
+	switch component {
+	case names.LighthouseCoreDNSComponent:
+		cmdArg = "--subm-version"
+	case names.MetricsProxyComponent:
+		cmdArg = "version"
+	}
+
+	execOptions := pods.ExecOptionsFromPod(pod)
+	execConfig := pods.ExecConfig{
+		RestConfig: clusterInfo.RestConfig,
+		ClientSet:  clusterInfo.ClientProducer.ForKubernetes(),
+	}
+
+	execOptions.Command = []string{componentCmd[component], cmdArg}
+
+	outStr, errStr, err := pods.ExecWithOptions(context.TODO(), execConfig, &execOptions)
+	if err != nil {
+		return "", errors.Wrap(err, errStr)
+	}
+
+	if component == names.MetricsProxyComponent {
+		return outStr, nil
+	}
+
+	result, found := strings.CutPrefix(errStr, fmt.Sprintf("%s version: ", component))
+
+	if !found {
+		return "", errors.Errorf("Failed to extract %s version from %s", component, errStr)
```

Comment:
I don't think we should return an error here - an empty version indicates not found.